### DISCUSSION TITLE
Volume buttons do not snooze

### DISCUSSION THREAD

**Discussion by Seithx** [NightscoutFoundation/xDrip#3754] [Bug reports]
Hi,

on google pixel 6a, not a new issue. It just does not work.

also, if I happen to be on my phone, and I slide the notification too fast, it also does not silence and I have to wait it out. I gotta wait 2 secs before I do.

I took no steps, just was like this for 2 years or more.

![Screenshot_20241103-185815](https://github.com/user-attachments/assets/a5cda0b8-65ed-404a-bde0-ddbd3a4b7b93)
![Screenshot_20241103-185856](https://github.com/user-attachments/assets/4b9f9d83-8d86-407c-9b4a-1e10053fd57c)

**Comment by Navid200** [comment]
The volume buttons silence the alert only if the screen is unlocked and xDrip is open.
Are you sure it doesn't work under those conditions?

I need to see what you mean by swiping the notification not snoozing the alert.
The only time I have experienced something unexpected was when I swiped the notification as soon as the alert triggered and instead of snoozing, it asked me to confirm.

I need to see how you swipe the notification and xDrip just ignores.

**Reply by Seithx** [reply]
Yes, I am sure. When the screen is locked, I unlock and it does not work, same if I am literally on the phone and not locked. positive. I go inside xdrip and disable it, sometimes enabling to be able to disable back again.

and OK, ill send this, though to be honest its very hard to replicate artificially, it usually happens when Im doing something else on my phone and when I hit the vol button it just lowers my volume & snoozing mostly does not work.

**Comment by corneliusroemer** [comment]
I've also noticed that alert snoozing doesn't seem to work anymore, at least if you use volume down button.

A toast will come up that alert is snoozed but it will fire again when the next reading comes in.

**Reply by Navid200** [reply]
OK, I think I see the problem.

This feature was added a very long time ago.
Since then, an optional button has been added inside the alert notification.

When I think about this, I see that we can spend time fixing this, or we can just press the button inside the notification button.
You cannot just press the button and expect it to snooze the alert under all conditions.  The phone must be unlocked and xDrip must be open.  But, under those same conditions, there will be a notification with a button right inside it.  It is as easy to tap that button if not easier than pressing a volume button.

So, I am not going to say the problem reported here does not exist.  But, I wonder do we really need to fix it?  Does it offer any value?

**Comment by georgebarnett** [comment]
I use a King Kong mini which is a small device with small tap targets. I have always appreciated the hardware buttons as they’re physical and can be felt in the dark or in a pocket.  It makes snoozing effortless.

**Reply by Navid200** [reply]
How can you unlock the phone and open xDrip in your pocket?
The snoozing function using volume buttons require the phone to be unlocked and the xDrip app being open.  It was never able to snooze by just pressing the volume button on a locked phone.

Considering that, I am not sure there is any benefit in fixing it.

**Reply by georgebarnett** [reply]
I understand your skepticism.  I find it easy to pickup and unlock my phone using muscle memory given the phone is small and the swipe code is the same.  From there xdrip+ is usually open and it’s a single click to mute and another to relock.  All of this works well for me without needing to visually inspect the tap target.  

I struggle to make the buttons in general work as they’re quite small.  I will of course make do if this isn’t fixed, but I would be grateful if a fix is considered.  Thank you.

**Reply by Navid200** [reply]
Very well.

I tried it on my Pixel 6a phone and can confirm that it doesn't snooze.

I don't know what has caused this.
I need to try an older phone to see if it works on it.

Sometimes, when we retire an Android level, some things change and it may take quite a while to notice some.  This could be one of those after we retired Android 6 quite a while ago.

**Reply by jamorham** [reply]
I used this just the other day but maybe it was on an older phone like Android 13 so it should still work. I can retest.

**Reply by rudysroost** [reply]
I used this last night on my Pixel 9 Pro XL running Android 16. It seems to be working correctly on my device.

**Reply by georgebarnett** [reply]
As would always happen in these cases, it's gone from not working to working and I am unable to reproduce. :(

**Reply by Seithx** [reply]
It's not working, I can assure you, on any on my phones it didnot - pixel 6a, pixel 10 (now). 
it's just bothersome during a meeting to go ''uhh, wait a second, my blood sugar'", then 2 mins I am trying to silence.
That's just the physical buttons, the software snoozing does the same thing basically, i swipe right, left, up - if it's right or very close to when the alert had started it just does not snooze, I have to disable all, then try again, and sometimes that does not always work.

**Reply by Navid200** [reply]
@Seithx I just tested with my Android 16 Pixel 6a and it did not work for me either.

What I find surprising about your description is that swiping does not snooze it.  I have never experienced that.  Of course, there are times I swipe and it asks me to approve.  That is not consistent.  And, it happens.  
When you want to snooze and it does not, have you tried using the main menu and tapping on "Snooze Alert"?
